rados/omap.go

```diff
@@ -105,6 +105,104 @@ func (ioctx *IOContext) SetOmap(oid string, pairs map[string][]byte) error {
 	return op.operateCompat(ioctx, oid)
 }
 
+// OmapKeyValue items are returned by the OmapGetElement's Next call.
+type OmapKeyValue struct {
+	Key   string
+	Value []byte
+}
+
+// OmapGetElement values are used to get the results of an GetOmapValues call
+// on a WriteOp. Until the Operate method of the WriteOp is called the Next
+// call will return an error. After Operate is called, the Next call will
+// return valid results.
+// The life cycle of the OmapGetElement is bound to the ReadOp, if the ReadOp
+// Release method is called the element may no longer be used.
+type OmapGetElement struct {
+	// inputs:
+	startAfter   string
+	filterPrefix string
+	maxReturn    uint64
+
+	// arguments:
+	cStartAfter   *C.char
+	cFilterPrefix *C.char
+
+	// C returned data:
+	iter C.rados_omap_iter_t
+	more C.uchar
+	rval C.int
+
+	// internal state:
+
+	// canIterate is only set after the operation is performed and is
+	// intended to prevent premature fetching of data from the element
+	canIterate bool
+}
+
+func newOmapGetElement(startAfter, filterPrefix string, maxReturn uint64) *OmapGetElement {
+	oge := &OmapGetElement{
+		startAfter:    startAfter,
+		filterPrefix:  filterPrefix,
+		maxReturn:     maxReturn,
+		cStartAfter:   C.CString(startAfter),
+		cFilterPrefix: C.CString(filterPrefix),
+	}
+	runtime.SetFinalizer(oge, freeElement)
+	return oge
+}
+
+func (oge *OmapGetElement) free() {
+	oge.reset()
+	C.free(unsafe.Pointer(oge.cStartAfter))
+	oge.cStartAfter = nil
+	C.free(unsafe.Pointer(oge.cFilterPrefix))
+	oge.cFilterPrefix = nil
+}
+
+func (oge *OmapGetElement) reset() {
+	if oge.canIterate {
+		C.rados_omap_get_end(oge.iter)
+	}
+	oge.canIterate = false
+	oge.more = 0
+	oge.rval = 0
+}
+
+func (oge *OmapGetElement) update() error {
+	oge.canIterate = true
```

Comment:
Not sure if there can be a  next element if the next line returns an error. But, I also can't remember when `update()` (and `reset()`) is called... Might be a hint that a comment about the function is needed?

Reply:
Good point. I'll look into this some more.

Reply:
This is still a TODO, I think?

Reply:
Yeah. Working on that part now.